src/main.rs

```diff
@@ -1,14 +1,51 @@
-use std::io::Read;
+use schemars::schema_for;
+use std::{env::args, fs, io::Read, process::exit};
 
 use nftables::schema::Nftables;
 
 fn main() {
-    deserialize_stdin();
+    // Get command arguments.
+    // We can skip the first argument in any case, because its the program path itself
+    let args = args().skip(1).collect();
+
+    handle_args(args);
+}
+
+fn handle_args(args: Vec<String>) {
+    let mut args = args.into_iter();
+    match args.next() {
+        Some(command) if command == "schema" => {
+            generate_json_schema(args.next().unwrap_or("./nftables.schema.json".to_string()));
+        }
+        Some(command) => {
+            println!("Unknown command: `{command}`. Try with schema to generate a Json schema or call with stdin only.");
+            exit(1);
+        }
+        // Called without any arguments so do what we did before already
+        None => {
+            deserialize_stdin();
+        }
+    }
+}
```

Comment:
I had the feeling that there might be some more potential for refactoring – what do you think about this? :)

```suggestion
    let command_args = collect_command_args();

    handle_args(command_args);
}

/// Get command arguments.
///
/// This skips the first argument, because it is the program path itself.
fn collect_command_args() -> Vec<String> {
    args().skip(1).collect()
}

fn handle_args(args: Vec<String>) {
    let command_arg = args.into_iter().next();

    if let Some(command) = &command_arg {
        if command == "schema" {
            generate_json_schema(command_arg.unwrap_or("./nftables.schema.json".to_string()));
            return;
        }

        println!("Unknown command: `{command}`. Try again with the schema command to generate a JSON Schema or call with stdin only.");
        exit(1);
    }

    deserialize_stdin();
}
```